cpp/src/arrow/type.h

```diff
@@ -76,7 +76,7 @@ class ARROW_EXPORT Fingerprintable {
   }
 
  protected:
-  Fingerprintable() = default;
+  Fingerprintable() : fingerprint_(NULLPTR), metadata_fingerprint_(NULLPTR) {}
```

Comment:
Instead, could you add the default initializers to the data members?
```c++
  mutable std::atomic<std::string*> fingerprint_{NULLPTR};
  mutable std::atomic<std::string*> metadata_fingerprint_{NULLPTR};
```

Reply:
Cool, wasn't aware that syntax existed in that context (just initialization via assignment, which doesn't work). 69b7b89